website/src/plugins/github-release-plugin.ts

```diff
@@ -0,0 +1,178 @@
+import type { Plugin } from '@docusaurus/types';
+import { Octokit } from '@octokit/rest';
+
+interface DownloadData {
+  version: string;
+}
+
+// --- Data type definitions for each OS ---
+
+export interface LinuxDownloadData extends DownloadData {
+  flatpak: string;
+  amd64: string;
+  arm64: string;
+}
+
+export interface MacosDownloadData extends DownloadData {
+  universal: string;
+  x64: string;
+  arm64: string;
+  airgapsetupX64: string;
+  airgapsetupArm64: string;
+}
+
+export interface WindowsDownloadData extends DownloadData {
+  binaryX64: string;
+  binaryArm64: string;
+  setupX64: string;
+  setupArm64: string;
+  airgapsetupX64: string;
+  airgapsetupArm64: string;
+}
+
+// --- Combined data types ---
+
+export interface GlobalData {
+  linuxDownloads: LinuxDownloadData;
+  macosDownloads: MacosDownloadData;
+  windowsDownloads: WindowsDownloadData;
+}
+
+// The main plugin function, now strongly typed
+export default async function githubReleasePlugin(): Promise<Plugin<GlobalData>> {
+  return {
+    name: 'docusaurus-plugin-github-release',
+
+    async loadContent(): Promise<GlobalData> {
+      console.log('Fetching latest GitHub release for podman-desktop using Octokit...');
+
+      const octokit = new Octokit();
+
+      try {
```

Comment:
fetching release should be a dedicated method (easier to have unit tests)

and easier when we would plug something like fetchStars() for https://github.com/podman-desktop/podman-desktop/issues/12452